e2e/kms.go

```diff
@@ -107,3 +108,68 @@ func (vc *vaultConfig) getPassphrase(f *framework.Framework, key string) (string
 
 	return strings.TrimSpace(stdOut), strings.TrimSpace(stdErr)
 }
+
+// creatEncSetup sets up required resources for encryption with given kmsID
+func createEncSetup(f *framework.Framework, kmsID string) {
+	switch kmsID {
```

Comment:
this is wrong, we don't want switch/cases for KMSs. Do something like this to make it more modular, and adding a new KMS will only need to implement its own functions that way:

```go
type kmsConfig interface {
   ...
   createSetup(f *framework.Framework) error
}

type simpleKMS struct {
    ...
    create func(f *framework.Framework) error
}

var (
        ...

	vaultTokensKMS = &vaultConfig{
		simpleKMS: &simpleKMS{
			provider: "vaulttokens",
                        createFunc: createVaultTokensKMS,
		},
		backendPath: defaultVaultBackendPath,
	}
        ...
}

func createVaultTokensKMS(f *framework.Framework) error {
    return createSecret(f.ClientSet, vaultExamplePath+"tenant-token.yaml", tenant)
}

func (sk *simpleKMS) createSetup(f *framework.Framework) error {
    if sk.createFunc != nil {
        return sk.createFunc(f)
    }
    return nil
}
```

This way you can just always call `.createSetup()` on the KMS instance that is going to be used in the test. The same object is already passed to validation functions.